DOC: Fix docstrings for Timestamp: unit, utcoffset, utctimetuple

shreyas-dev opened the pull request:
Part of https://github.com/pandas-dev/pandas/issues/59458

Addresses:

pandas.Timestamp.unit having SA01
pandas.Timestamp.utcoffset having SA01
pandas.Timestamp.utctimetuple having SA01


Changes:
Added See Also section for all 3.
Added Returns section for all 3.
Added extended section for all 3 
Removed all 3 from code_checks.sh

rob-sil committed:
DOC: Add missing single quote to `Timestamp.tz_localize` (#59472)

* Add missing single quote to  docstring

* More missing single quotes

* Clean up a print statement

Update docstring for utcoffset

shreyas-dev committed:
Update docstring for utcoffset

shreyas-dev committed:
Update Docstrings for unit, utcoffset, utctimetuple

Use old unit one-line description

Remove pandas prefix from See Also due to SA05

shreyas-dev committed:
Update Docstrings for unit, utcoffset, utctimetuple

shreyas-dev committed:
Use old unit one-line description

shreyas-dev committed:
Remove pandas prefix from See Also due to SA05

shreyas-dev committed:
Remove unit, utcoffset, utctimetuple from code checks

shreyas-dev committed:
Remove unit, utcoffset, utctimetuple from code checks

shreyas-dev committed:
Remove unit, utcoffset, utctimetuple from code checks

Add backticks

Add new line for extended summary in utcoffset

shreyas-dev committed:
Add backticks

shreyas-dev committed:
Add new line for extended summary in utcoffset

luke396 committed:
DOC: More doc fix for dtype_backend (#59071)

awojno-bloomberg committed:
DOC: Enforce Numpy Docstring Validation for pandas.Series.str methods (#59474)

* Doctrings for pandas.Series.str functions
    pandas.Series.str.capitalize
    pandas.Series.str.casefold
    pandas.Series.str.center
    pandas.Series.str.decode
    pandas.Series.str.encode

* Fixed other docstrings along the way

mroeschke reviewed (changes requested):
You'll need to fix the NaT docstrings as well

jorisvandenbossche committed:
BUG (string dtype): convert dictionary input to materialized string array in ArrowStringArray constructor (#59479)

bmarika committed:
Correct typo (#59486)

partev committed:
use x.com and twitter's new X logo (#59482)

jorisvandenbossche committed:
String dtype: fix convert_dtypes() to convert NaN-string to NA-string (#59470)

* String dtype: fix convert_dtypes() to convert NaN-string to NA-string

* fix CoW tracking for conversion to python storage strings

* remove xfails

johnyu013 committed:
DOC: Clarify nrows behavior in read_csv (#59467)

* DOC: Clarify nrows behavior in read_csv

* Remove whitespace from a blank line

mroeschke committed:
STY: Add Yields section to option_context (#59490)

Add Yields section to option_context

jsspencer committed:
BUG: Fix out-of-bounds access in safe_sort with an empty array and non-empty codes (#59489)

* Fix out-of-bounds violations in safe_sort for empty arrays.

Previously we masked `codes` referring to out-of-bounds elements to 0 and
then fixed them after to -1 using `np.putmask`. However, this results in
out-of-bounds access in `take_nd` if the array is empty.

Instead, set all out-of-bounds indices in `codes` to -1 immediately, as
these can be handled by `take_nd`.

* Remove dead code.

`use_na_sentinel` cannot be truthy inside an else branch where it is
falsy.

* Add test based upon #59421

MichaelMoyles committed:
DOC: Fix docstrings for Timestamp: tz_localize, tzname, utcfromtimestamp (#59484)

* updated docstring for pandas.Timestamp.tz_localize, pandas.Timestamp.tzname and pandas.Timestamp.utcfromtimestamp

* trimmed whitespace

* reducing line length

* updating NaT docstrings to match for timestamps

mroeschke committed:
DEPS: Make pytz an optional dependency (#59089)

* Make pytz an optional dependency

* Start to address tests

* Fix tests

* Fix tests

* Fix test, import optional pytz in conftest

* Fix formatting

* Change minimum

* remove type ignore

* another pa under 17

* Address comments

* Undo file

* Fix pyarrow 17 test

* Test xpasses on pyarrow 18

jorisvandenbossche committed:
String dtype: honor mode.string_storage option (and change default to None) (#59488)

* String dtype: honor mode.string_storage option (and change default to None)

* fix test + explicitly test default

* use 'auto' instead of None

tswast committed:
DOC: add pandas-gbq and related db-dtypes packages to ecosystem (#59491)

* DOC: add pandas-gbq and related db-dtypes packages to ecosystem

As of pandas 2.2.0, pandas-gbq functionality is fully separated, so I think now would be a good time to introduce it as part of the "ecosystem".

db-dtypes was built for use from pandas-gbq but should be applicable more generally.

* fix pandas-gbq link

shreyas-dev committed:
Update nattype.pyx to match updated docstrings

shreyas-dev commented:
Will raise another PR
